src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -63,13 +63,13 @@ private BoundStatement MakeLoweredForm(BoundPatternSwitchStatement node)
                     initialTemp = _factory.SynthesizedLocal(expression.Type, expression.Syntax);
                     result.Add(_factory.Assignment(_factory.Local(initialTemp), expression));
                     expression = _factory.Local(initialTemp);
-                }
 
-                // EnC: We need to insert a hidden sequence point to handle function remapping in case 
-                // the containing method is edited while methods invoked in the expression are being executed.
-                if (!node.WasCompilerGenerated && _localRewriter.Instrument)
-                {
-                    expression = _localRewriter._instrumenter.InstrumentSwitchStatementExpression(node, expression, _factory);
+                    // EnC: We need to insert a hidden sequence point to handle function remapping in case 
+                    // the containing method is edited while methods invoked in the expression are being executed.
+                    if (!node.WasCompilerGenerated && _localRewriter.Instrument)
+                    {
+                        expression = _localRewriter._instrumenter.InstrumentSwitchStatementExpression(node, expression, _factory);
```

Comment:
I've opened #17090 to track that.

---
In reply to: [100597429](https://github.com/dotnet/roslyn/pull/16928#discussion_r100597429) [](ancestors = 100597429)